crypto/ui/ui_util.c

```diff
@@ -109,7 +104,8 @@ static int ui_read(UI *ui, UI_STRING *uis)
             char result[PEM_BUFSIZE];
             const struct pem_password_cb_data *data =
                 UI_method_get_ex_data(UI_get_method(ui),
-                                      ui_method_data_index());
+                                      RUN_ONCE(&get_index_once,
+                                               ui_method_data_index));
```

Comment:
> The call in ```UI_UTIL_wrap_read_pem_callback()``` will always happen first

I sort-of suspected that was the case, but I'm still a little foggy after being out sick yesterday, so wanted to double-check.

I guess we should let Rich resolve his "a bit tricky" remark before you merge it, though.